src/v/serde/parquet/encoding.cc

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "serde/parquet/encoding.h"
+
+#include <algorithm>
+#include <climits>
+
+namespace serde::parquet {
+
+iobuf encode_plain(const chunked_vector<boolean_value>& vals) {
+    iobuf buf;
+    size_t i = 0;
+    while (i < vals.size()) {
+        size_t pass_size = std::min<size_t>(vals.size() - i, CHAR_BIT);
+        uint8_t tmp = 0;
+        for (; i < pass_size; ++i) {
+            tmp >>= 1U;
+            static constexpr uint8_t leading_bit = 0b1000'0000;
+            tmp |= vals[i].val ? leading_bit : 0U;
+        }
+        buf.append(&tmp, 1);
+    }
+    return buf;
+}
+
+iobuf encode_plain(const chunked_vector<int32_value>& vals) {
+    iobuf buf;
+    for (const auto& v : vals) {
+        int32_t i = ss::cpu_to_le(v.val);
+        // NOLINTNEXTLINE(*reinterpret-cast*)
+        buf.append(reinterpret_cast<const uint8_t*>(&i), sizeof(int32_t));
+    }
+    return buf;
+}
+
+iobuf encode_plain(const chunked_vector<int64_value>& vals) {
+    iobuf buf;
+    for (const auto& v : vals) {
+        int64_t i = ss::cpu_to_le(v.val);
+        // NOLINTNEXTLINE(*reinterpret-cast*)
+        buf.append(reinterpret_cast<const uint8_t*>(&i), sizeof(int64_t));
+    }
+    return buf;
+}
+
+iobuf encode_plain(const chunked_vector<float32_value>& vals) {
+    iobuf buf;
+    for (const auto& v : vals) {
+        // NOLINTNEXTLINE(*reinterpret-cast*)
+        buf.append(reinterpret_cast<const uint8_t*>(&v.val), sizeof(float));
+    }
+    return buf;
+}
+
+iobuf encode_plain(const chunked_vector<float64_value>& vals) {
+    iobuf buf;
+    for (const auto& v : vals) {
+        // NOLINTNEXTLINE(*reinterpret-cast*)
+        buf.append(reinterpret_cast<const uint8_t*>(&v.val), sizeof(double));
+    }
+    return buf;
+}
+
+iobuf encode_plain(chunked_vector<byte_array_value> vals) {
+    iobuf buf;
+    for (auto& v : vals) {
+        int32_t i = ss::cpu_to_le(static_cast<int32_t>(v.val.size_bytes()));
+        // NOLINTNEXTLINE(*reinterpret-cast*)
+        buf.append(reinterpret_cast<const uint8_t*>(&i), sizeof(int32_t));
+        buf.append(std::move(v.val));
+    }
+    return buf;
+}
+
+iobuf encode_plain(chunked_vector<fixed_byte_array_value> vals) {
+    iobuf buf;
+    for (auto& v : vals) {
+        buf.append(std::move(v.val));
```

Comment:
where does the protocol record the size of the byte array?

Reply:
It's in the schema (which is in file metadata). The byte array type in the schema with an optional fixed length